jobs/periodic/flag-old-conversations/index.ts

```diff
@@ -0,0 +1,94 @@
+    return !dissolvedAtPlus30Days.isBefore(today);
+}
+
+async function isActiveSubcourse(id: number): Promise<boolean> {
+    const today = moment().endOf('day');
+    const subcourse = await getSubcourse(id, true);
+    const isSubcourseCancelled = subcourse.cancelled;
+
+    if (isSubcourseCancelled) {
+        return false;
+    } else {
+        const lastLecutre = subcourse.lecture.sort((a: Lecture, b: Lecture) => moment(a.start).milliseconds() - moment(b.start).milliseconds()).pop();
+        const lastLecturePlus30Days = moment(lastLecutre.start).add(30, 'days');
+        const is30DaysBeforeToday = lastLecturePlus30Days.isBefore(today);
+        return !is30DaysBeforeToday;
+    }
+}
+
+export default async function flagInactiveConversationsAsReadonly() {
+    const conversations = await getAllConversations();
+    const conversationsTo: conversationsToDeactivate[] = [];
+
+    for (const conversation of conversations.data) {
+        let shouldMarkAsReadonly: boolean;
+        if (conversation.custom.subcourse) {
+            const subcourseIds: number[] = JSON.parse(conversation.custom.subcourse);
+            const allSubcoursesActive = await Promise.all(subcourseIds.map((id) => isActiveSubcourse(id)));
+            shouldMarkAsReadonly = allSubcoursesActive.every((active) => active === false);
+        }
+        if (conversation.custom.prospectSubcourse) {
+            const prospectSubcourses: number[] = conversation.custom.prospectSubcourse;
+            const allProspectSubcoursesActive = await Promise.all(prospectSubcourses.map((id) => isActiveSubcourse(id)));
+            shouldMarkAsReadonly = allProspectSubcoursesActive.every((active) => active === false);
+        }
+
+        if (conversation.custom.match) {
+            const match = JSON.parse(conversation.custom.match);
+            const matchId = match.matchId;
+            const isMatchActive = await isActiveMatch(matchId);
+            shouldMarkAsReadonly = !isMatchActive;
+        }
+
+        if (shouldMarkAsReadonly) {
+            if (conversation.custom.match) {
+                conversationsTo.push({
+                    id: conversation.id,
+                    conversationType: ConversationType.ONE_ON_ONE,
+                });
+            } else {
+                conversationsTo.push({
+                    id: conversation.id,
+                    conversationType: ConversationType.GROUP,
+                });
+            }
+        }
+    }
+
+    if (conversationsTo.length > 0) {
+        for (const convo of conversationsTo) {
+            await markConversationAsReadOnly(convo.id);
+            if (convo.conversationType === ConversationType.ONE_ON_ONE) {
+                await sendSystemMessage(systemMessages.de.deactivated, convo.id, SystemMessage.ONE_ON_ONE_OVER);
+            } else {
+                await sendSystemMessage(systemMessages.de.deactivated, convo.id, SystemMessage.GROUP_OVER);
+            }
```

Comment:
To get some more information about which conversations have been marked as RO. This could be useful for debugging later on.
```suggestion
            }
            logger.info('Mark converstation as readonly', {conversation_id: convo.id, conversation_type: convo. conversationType})
```